UniWatch/Controllers/ClassController.cs

```diff
@@ -0,0 +1,129 @@
+        {
+            var teacher = _dataAccess.UserManager.GetTeacherById(teacherId);
+                if (teacher==null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+            ViewBag.TeacherId = teacherId;
+            return View(_dataAccess.ClassManager.GetClassesForTeacher(teacherId));
+        }
+
+
+        // GET: Create
+        /// <summary>
+        /// Sends user to the create class page
+        /// </summary>
+        /// <param name="teacherId">Id of the teacher</param>
+        public ActionResult Create(int teacherId)
+        {
+            var teacher = _dataAccess.UserManager.GetTeacherById(teacherId);
+            if (teacher == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+           // var teacher = _dataAccess.UserManager.GetTeacherById(teacherId);
+            return View(new Class() { Teacher = teacher });
+        }
+
+        //POST: Create
+        /// <summary>
+        /// Create class for the teacher
+        /// </summary>
+        /// <param name="class">Model containing information to create the class</param>
+        /// <param name="teacherId">Id of the teacher</param>
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult Create([Bind(Include = "Name,Number,Section,Semester,Year")] Class @class, int teacherId)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View(@class);
+            }
+            try
+            {
+                var created = _dataAccess.ClassManager.CreateClass(@class.Name, @class.Number, @class.Section, @class.Semester, @class.Year, teacherId);
+                return RedirectToAction("Index", new { teacherId = created.Teacher.Id });
+            }
+            catch (InvalidOperationException)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.Conflict);
+            }
+          
+        }
+
+        //GET: Delete
+        /// <summary>
+        /// Sends user to the delete class page
+        /// </summary>
+        /// <param name="classId">Id of the class</param>
+        public ActionResult Delete(int classId)
+        {
+            var @class = _dataAccess.ClassManager.GetById(classId);
+            if (@class == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest, $"No class with id {classId}");
+            }
+
+            return View(@class);
+        }
+
+        //POST: Delete
+        /// <summary>
+        /// Deletes the class
+        /// </summary>
+        /// <param name="classId"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [ActionName("Delete")]
+        [ValidateAntiForgeryToken]
+        public ActionResult DeleteConfirmed(int classId)
+        {
+            var @class = _dataAccess.ClassManager.GetById(classId);
+            var teacherId = @class.Teacher.Id;
```

Comment:
moving that causes @class to error on the if statement since its not declared within its parameters. Are you just saying to move var teacherID information?